src/Application/ConfigResolver.php

```diff
@@ -62,7 +64,7 @@ public static function resolve(array $config, InputInterface $input): Configurat
 
         $isRootAnalyse = true;
         foreach (Kernel::getRequiredFiles() as $file) {
-            if (! file_exists($directory . DIRECTORY_SEPARATOR . $file)) {
+            if (! file_exists($directories[0] . DIRECTORY_SEPARATOR . $file)) {
                 $isRootAnalyse = false;
```

Comment:
Well, if I remember correctly, `$isRootAnalyse` is used to exclude some global insights related to composer. Kernel::getRequiredFiles return [composer.json, composer.lock] 

As you manage the possibility to pass a custom composer.json, maybe it should be handle here ?

Reply:
Nice catch! Done.